server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -295,7 +295,9 @@ public PostingsFormat getPostingsFormat(String field) {
     }
 
     void checkLimits(IndexSettings settings) {
-        checkFieldLimit(settings.getMappingTotalFieldsLimit());
+        if (settings.isIgnoreDynamicFieldsBeyondLimit() == false) {
```

Comment:
**NOTE** that I had to change this because of the way settings are composed. In other places we also always check `isIgnoreDynamicBeyondFieldLimit` before actually checking the limit.

Reply:
This needs to be applied only to dynamically mapped fields anyway.